### ISSUE TITLE
Can't use geth as a client (web3) with OpenEthereum 3.1

### ISSUE THREAD

**Issue by gituser** [openethereum/openethereum#176]
For questions please use https://discord.io/openethereum, issues are for bugs and feature requests.

_Before filing a new issue, please **provide the following information**._

- **OpenEthereum version (>=3.1.0)**: 3.1.0
- **Operating system**: Linux
- **Installation**:  built from source
- **Fully synchronized**: yes
- **Network**: ethereum / ropsten (doesn't matter)
- **Restarted**: yes

On previous versions I was able to use `geth` as a client to OpenEthereum / Parity.

But not anymore geth gives me this error:
```
$ $GETH --exec "eth.blockNumber" attach ipc:$IPCPATH
Fatal: Failed to start the JavaScript console: api modules: Method not found
```
I've been using geth to query the daemon for e.g. `eth.blockNumber` among other things from the console instead of RPC.

It seems the `api` module was removed entirely from new version. 
Any ideas how to fix this?

**Comment by smartins** [comment]
Getting the same error with 3.1.1

**Comment by pwagner** [comment]
I get the same error

> Fatal: Failed to start the JavaScript console: api modules: Method not found

trying to run `geth attach ws://localhost:8546` (and `geth attach http://localhost:8545` etc.) with
- version OpenEthereum/v3.1.1-stable-15b558189-20210119/x86_64-macos/rustc1.49.0
- version OpenEthereum/v3.1.0-stable-94a784b-20201106/x86_64-macos/rustc1.47.0

Operating system: macOS Catalina
Installation: 3.1.1 built from source, 3.1.0 via homebrew
Fully synchronized: yes
Network: mainnet
Restarted: yes
(geth version 1.9.25-stable)

**Comment by gituser** [comment]
ping @rakita

**Comment by kf106** [comment]
Is there an ETA on fixing this?

**Comment by rakita** [comment]
@gituser @kf106 can you test it, it is merged in `dev` branch? If there is a problem reopen this issue

**Comment by gituser** [comment]
@rakita 
I've tried `parity-v3.2.0-rc.1-stable-0947261-20210226` built from `dev` branch the issue is still there:

```
Fatal: Failed to start the JavaScript console: api modules: Method not found
```

**Comment by gituser** [comment]
@rakita 
It seems in the `v3.2.0` this fix never applied https://github.com/openethereum/openethereum/pull/284

Can you release `v3.2.1` with this fix included?

Thank you.

**Comment by gituser** [comment]
ping @rakita @sunce86

**Comment by rakita** [comment]
I want to have only one version for Berlin hard fork and don't potentially introduce confusion with v3.2.1. In next iteration that will be relatively soon, just after berlin is finished, this will be included.

**Comment by gituser** [comment]
@rakita I get it ok.

Just for those who are migrating from `v2.5.13` this issue will be breaking if they are using any socket console interaction with the node and not just RPC calls.

When's the fork is scheduled?

**Comment by rakita** [comment]
@gituser Ropsten was forked yesterday. Here is full timeline: https://blog.ethereum.org/2021/03/08/ethereum-berlin-upgrade-announcement/

**Comment by gituser** [comment]
@rakita I see you pushed a new tag https://github.com/openethereum/openethereum/releases/tag/v3.2.1 with hot-fix, maybe you could also push another one with this fix included?

Judging from this thread others will appreciate this fix.

Thank you.

**Comment by gaia** [comment]
Found this thread searching for "Failed to start the JavaScript console: api modules: Method not found". To help others that came here for the same reason:

1. Will this be fixed in the future?*
2. Is RPC the only alternative to a (geth like) console?

* changing `apis=` in `[rpc]` from the current `apis = ["web3", "eth", "net", "parity", "traces"]` to the suggested `apis = ["web3", "eth", "pubsub", "net", "parity", "parity_pubsub", "traces", "personal", "parity_accounts", "parity_set", "rpc"]` prevents OE from starting

**Comment by vlddm** [comment]
Same issue with `OpenEthereum/v3.2.4-stable-0996732-20210415/x86_64-linux-gnu/rustc1.51.0`
If openethereum running with `--ipc-apis=all` got this error:
```
geth attach .local/share/openethereum/jsonrpc.ipc 
Fatal: Failed to start the JavaScript console: api modules: Method not found
```

Trying to start OE with `--ipc-apis=rpc`:
```
Unknown api: rpc
Error: 1
```

**Comment by gituser** [comment]
@vlddm, you need to build from source that latest tag and cherry pick this pull request https://github.com/openethereum/openethereum/pull/284

```
git checkout v3.2.4
git cherry-pick ed12fffeeffc12093d4313fe7082b7049e287b5f
cargo build --release --features final
```

Most likely this will be included in `v3.2.5`

**Comment by gituser** [comment]
`v3.2.5` has been released with the fix included - https://github.com/openethereum/openethereum/releases/tag/v3.2.5
